tests/unit/ssg-module/test_id_translate.py

```diff
@@ -21,3 +26,172 @@ def test__split_namespace():
     ns, n = sn("{}emptynamespace")
     assert not ns
     assert n == "emptynamespace"
+
+
+def test_tagname_to_abbrev():
+    ab = ssg.id_translate._tagname_to_abbrev(
+        "{%s}definition" % oval_namespace)
+    assert ab == "def"
+    ab = ssg.id_translate._tagname_to_abbrev("{%s}criteria" % oval_namespace)
+    assert ab == "crit"
+    ab = ssg.id_translate._tagname_to_abbrev("{%s}test" % oval_namespace)
+    assert ab == "tst"
+    ab = ssg.id_translate._tagname_to_abbrev("{%s}object" % oval_namespace)
+    assert ab == "obj"
+    ab = ssg.id_translate._tagname_to_abbrev("{%s}state" % oval_namespace)
+    assert ab == "ste"
+    ab = ssg.id_translate._tagname_to_abbrev("{%s}variable" % oval_namespace)
+    assert ab == "var"
+
+
+def test_tagname_to_abbrev_unknown_ns():
+    with pytest.raises(RuntimeError):
+        unknown_name_space = "http://world.wide.web/some/unknown/foo.php"
+        ssg.id_translate._tagname_to_abbrev(
+            "{%s}rationale" % unknown_name_space)
+
+
+def test_tagname_to_abbrev_unknown_tag():
+    with pytest.raises(KeyError):
+        ssg.id_translate._tagname_to_abbrev("{%s}candy" % oval_namespace)
+    with pytest.raises(KeyError):
+        ssg.id_translate._tagname_to_abbrev("{%s}pizza" % ocil_namespace)
+
+
+@pytest.fixture
+def idtranslator():
+    return ssg.id_translate.IDTranslator("ssg")
+
+
+def test_idtranslator_generate_id(idtranslator):
+    genid = idtranslator.generate_id(
+        "{%s}object" % oval_namespace, "my_terrible_object")
+    assert genid == "oval:ssg-my_terrible_object:obj:1"
+    genid = idtranslator.generate_id(
+        "{%s}question" % ocil_namespace, "my_terrible_face")
+    assert genid == "ocil:ssg-my_terrible_face:question:1"
+
+
+@pytest.fixture
+def oval_tree():
+    draft_oval_path = os.path.join(DATADIR, "draft_oval.xml")
+    return ET.parse(draft_oval_path).getroot()
+
+
+def test_idtranslator_translate_oval(idtranslator, oval_tree):
```

Comment:
I see that this test takes an OVAL tree, and "translates" it, whatever it means.
I would expect this test to focus on expected differences between original and translated (or expected lack of differences). Instead, the test is mainly about making sure that the translated tree has certain properties, and it does so without referencing the original.